File: core/src/main/java/com/benny/openlauncher/core/util/BaseIconProvider.java
package com.benny.openlauncher.core.util;

import android.widget.TextView;

import com.benny.openlauncher.core.interfaces.IconDrawer;
import com.benny.openlauncher.core.interfaces.IconProvider;

public abstract class BaseIconProvider implements IconProvider {

    public void loadIconIntoIconDrawer(IconDrawer iconDrawer, int forceSize, int index) {
        loadIcon(IconTargetType.IconDrawer, forceSize, iconDrawer, index);
    }

    public void cancelLoad(IconDrawer iconDrawer) {
        cancelLoad(IconTargetType.IconDrawer, iconDrawer);
    }

    public void loadIconIntoTextView(TextView tv, int forceSize, int gravity) {
        loadIcon(IconTargetType.TextView, forceSize, tv, gravity);
    }

    public void cancelLoad(TextView tv) {
        cancelLoad(IconTargetType.TextView, tv);
    }
}

File: core/src/main/java/com/benny/openlauncher/core/util/BaseIconProvider.kt
package com.benny.openlauncher.core.util

import android.widget.TextView

import com.benny.openlauncher.core.interfaces.IconDrawer
import com.benny.openlauncher.core.interfaces.IconProvider

abstract class BaseIconProvider : IconProvider {

    fun loadIconIntoIconDrawer(iconDrawer: IconDrawer, forceSize: Int, index: Int) {
        loadIcon(IconProvider.IconTargetType.IconDrawer, forceSize, iconDrawer, index)
    }

    fun cancelLoad(iconDrawer: IconDrawer) {
        cancelLoad(IconProvider.IconTargetType.IconDrawer, iconDrawer)
    }

    fun loadIconIntoTextView(tv: TextView, forceSize: Int, gravity: Int) {
        loadIcon(IconProvider.IconTargetType.TextView, forceSize, tv, gravity)
    }

    fun cancelLoad(tv: TextView) {
        cancelLoad(IconProvider.IconTargetType.TextView, tv)
    }
}
